tests/linearizability/watch.go

```diff
@@ -82,28 +82,37 @@ func validateWatchResponses(t *testing.T, responses [][]watchResponse, expectPro
 }
 
 func validateMemberWatchResponses(t *testing.T, responses []watchResponse, expectProgressNotify bool) {
-	var gotProgressNotify = false
-	var lastRevision int64 = 1
+	var (
+		gotProgressNotify       = false
+		lastEventRevision int64 = 1 // The event.Kv.ModRevision in the latest event.
+		lastHeadRevision  int64 = 1 // The resp.Header.Revision in last watch response.
+	)
 	for _, resp := range responses {
-		if resp.Header.Revision < lastRevision {
-			t.Errorf("Revision should never decrease")
+		if resp.Header.Revision < lastEventRevision || resp.Header.Revision < lastHeadRevision {
+			t.Errorf("Revision should never decrease[1], lastEventRevision: %d, lastHeadRevision: %d, resp.Header.Revision: %d",
+				lastEventRevision, lastHeadRevision, resp.Header.Revision)
 		}
-		if resp.IsProgressNotify() && resp.Header.Revision == lastRevision {
+
+		if resp.IsProgressNotify() && resp.Header.Revision == lastHeadRevision {
 			gotProgressNotify = true
 		}
-		if resp.Header.Revision == lastRevision && len(resp.Events) != 0 {
+		if resp.Header.Revision == lastHeadRevision && len(resp.Events) != 0 {
 			t.Errorf("Got two non-empty responses about same revision")
 		}
+
 		for _, event := range resp.Events {
-			if event.Kv.ModRevision != lastRevision+1 {
-				t.Errorf("Expect revision to grow by 1, last: %d, mod: %d", lastRevision, event.Kv.ModRevision)
+			if event.Kv.ModRevision != lastEventRevision+1 {
```

Comment:
What if multiple KVs are changed within a single transaction ? 

I assume its possible to emit multiple events at the same events.Kv.ModRevision.

Reply:
Exactly, I raised [similar comment ](https://github.com/etcd-io/etcd/pull/15250#discussion_r1099327154) in another PR. It's fine since we do not support multi-operations TXN so far. Let's fix this in the PR which supports multi-operation TXN.

Reply:
Usually I tend to resolve issue step by step. I am fixing the issue what we have so far. Please let me know if you have big concern on this. thx

Reply:
Have a ready PR that will add multi op TXN https://github.com/etcd-io/etcd/pull/15259
Let's address it there.